conferences/ticketing_storage.go

```diff
@@ -0,0 +1,605 @@
+package conferences
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"encore.dev/storage/sqldb"
+)
+
+// SQLStorage provides a Postgres Flavored storage backend to store ticketing information.
+type SQLStorage struct {
+	tx *sqldb.Tx
+}
+
+const (
+	ticketIDUniqueConstraint = "ticket_id_is_unique"
+	tableSlotClaims          = "slot_claim"
+)
+
+func query(ctx context.Context, tx *sqldb.Tx, statement string, args ...interface{}) (*sqldb.Rows, error) {
+	if tx != nil {
+		return sqldb.QueryTx(tx, ctx, statement, args...)
+	}
+	return sqldb.Query(ctx, statement, args...)
+}
+
+func queryRow(ctx context.Context, tx *sqldb.Tx, statement string, args ...interface{}) *sqldb.Row {
+	if tx != nil {
+		return sqldb.QueryRowTx(tx, ctx, statement, args...)
+	}
+	return sqldb.QueryRow(ctx, statement, args...)
+}
+
+func exec(ctx context.Context, tx *sqldb.Tx, statement string, args ...interface{}) (sql.Result, error) {
+	if tx != nil {
+		return sqldb.ExecTx(tx, ctx, statement, args...)
+	}
+	return sqldb.Exec(ctx, statement, args...)
+}
+
+// createAttendee creates a new attendee in the database and returns it.
+func createAttendee(ctx context.Context, tx *sqldb.Tx, a *Attendee) (*Attendee, error) {
+	result := Attendee{}
+	row := queryRow(ctx, tx,
+		"INSERT INTO attendee (email, coc_accepted) VALUES ($1, $2) RETURNING id, email, coc_accepted",
+		a.Email, a.CoCAccepted)
+
+	if row == nil {
+		return nil, fmt.Errorf("new attendee was not created")
+	}
+
+	if err := row.Scan(&result.ID, &result.Email, &result.CoCAccepted); err != nil {
+		return nil, fmt.Errorf("creating or fetching attendee: %w", err)
+	}
+
+	newClaims := make([]SlotClaim, len(a.Claims))
+	for i := range a.Claims {
+		c := a.Claims[i]
+		res, err := exec(ctx, tx,
+			`INSERT INTO attendee_to_slot_claims (attendee_id, slot_claim_id) 
+			VALUES ($1, $2) 
+		ON CONFLICT ON CONSTRAINT slot_claim_id_is_unique DO UPDATE SET attendee_id = $3`,
+			result.ID, c.ID, result.ID)
```

Comment:
You can use `EXCLUDED` to reference rows that you were trying to insert to. 

```suggestion
		ON CONFLICT ON CONSTRAINT slot_claim_id_is_unique DO UPDATE SET attendee_id = EXCLUDED.attendee_id`,
			result.ID, c.ID)
```